accord-core/src/test/java/accord/KeysTest.java

```diff
@@ -225,6 +237,51 @@ public void slice()
         });
     }
 
+    @Test
+    public void keysWithout()
+    {
+        without(IntKey::key, Key[]::new, Keys::of, Keys::without);
+    }
+
+    @Test
+    public void routingKeysWithout()
+    {
+        without(IntKey::routing, RoutingKey[]::new, RoutingKeys::of, RoutingKeys::without);
+    }
+
+    private <K extends RoutableKey, KS extends AbstractKeys<K>> void without(IntFunction<K> keyFactory,
+                                                                             IntFunction<K[]> arrayFactory,
+                                                                             Function<K[], KS> keysFactory,
+                                                                             BiFunction<KS, KS, Routables<K>> without)
+    {
+        final int maxSize = 100;
+        final int maxKey = 1000;
+        for (int test = 0 ; test < 10000 ; ++test)
+        {
+            RandomTestRunner.test().check(rs -> {
+                K[] keys = arrayFactory.apply(rs.nextInt(2, maxSize));
+                IntHashSet used = new IntHashSet();
+                for (int i = 0 ; i < keys.length ; ++i)
+                {
+                    int k = rs.nextInt(maxKey);
+                    while (!used.add(k)) k = rs.nextInt(maxKey);
+                    keys[i] = keyFactory.apply(k);
+                }
+                int addTo = rs.nextInt(1, keys.length);
+                int removeFrom = addTo - rs.nextInt(addTo);
```

Comment:
```suggestion
                int removeFrom = addTo - rs.nextInt(addTo + 1);
```

this test doesn't cover the case that the remove set is fully inclusive of the add set, so never reaches the empty set.  By doing `addTo + 1` we allow `removeFrom` to reach `0` which would test the empty set as well.